tests/csapi/room_versions_test.go

```diff
@@ -0,0 +1,193 @@
+				body := gjson.ParseBytes(must.ParseJSON(t, res.Body))
+				defer res.Body.Close()
+				if len(body.Get("chunk").Array()) != 6 {
+					t.Fatal("Expected 6 messages")
+				}
+			})
+
+			// sytest: Can reject invites over federation for rooms with version $version
+			t.Run(fmt.Sprintf("Can reject invites over federation for rooms with version %s", roomVersion), func(t *testing.T) {
+				t.Parallel()
+				roomID := createRoomSynced(t, alice, map[string]interface{}{
+					"room_version": roomVersion,
+					"invite":       []string{charlie.UserID},
+				})
+				charlie.MustSyncUntil(t, client.SyncReq{}, client.SyncInvitedTo(charlie.UserID, roomID))
+				charlie.LeaveRoom(t, roomID)
+			})
+
+			// sytest: Can receive redactions from regular users over federation in room version $version
+			t.Run(fmt.Sprintf("Can receive redactions from regular users over federation in room version %s", roomVersion), func(t *testing.T) {
+				t.Parallel()
+				roomID := createRoomSynced(t, alice, map[string]interface{}{
+					"room_version": roomVersion,
+					"invite":       []string{charlie.UserID},
+				})
+				charlie.MustSyncUntil(t, client.SyncReq{}, client.SyncInvitedTo(charlie.UserID, roomID))
+				joinRoomSynced(t, charlie, roomID, "")
+				eventID := sendMessageSynced(t, charlie, roomID)
+				// redact the message
+				res := charlie.MustDoFunc(t, "POST", []string{"_matrix", "client", "v3", "rooms", roomID, "redact", eventID}, client.WithRawBody([]byte("{}")))
+				js := must.ParseJSON(t, res.Body)
+				defer res.Body.Close()
+				redactID := must.GetJSONFieldStr(t, js, "event_id")
+				alice.MustSyncUntil(t, client.SyncReq{}, client.SyncTimelineHas(roomID, func(result gjson.Result) bool {
+					return redactID == result.Get("event_id").Str
+				}))
+				// query messages
+				queryParams := url.Values{}
+				queryParams.Set("dir", "b")
+				res = alice.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "messages"}, client.WithQueries(queryParams))
+				body := gjson.ParseBytes(must.ParseJSON(t, res.Body))
+				defer res.Body.Close()
+				events := body.Get("chunk").Array()
+				// first event should be the redaction
+				must.EqualStr(t, events[0].Get("event_id").Str, redactID, "wrong event")
+				must.EqualStr(t, events[0].Get("redacts").Str, eventID, "wrong event")
+				// second event should be the original event
+				must.EqualStr(t, events[1].Get("event_id").Str, eventID, "wrong event")
+				must.EqualStr(t, events[1].Get("unsigned.redacted_by").Str, redactID, "wrong event")
+			})
+		}
+	})
+}
+
+func sendMessageSynced(t *testing.T, cl *client.CSAPI, roomID string) (eventID string) {
+	return cl.SendEventSynced(t, roomID, b.Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"msgtype": "m.text",
+			"body":    "hello world",
+		},
+	})
+}
+
+func joinRoomSynced(t *testing.T, cl *client.CSAPI, roomID, alias string) {
```

Comment:
These should probably go into the Client impl to be honest. For now leave them here though.